### PR TITLE
51172 - Choose a date error includes broken link to request flow

### PR DESCRIPTION
**Pull request by vbahinwillit**
## Summary
This PR fixes the message displayed when an error occurs retrieving appointment slots and the clinic is not configured to accept appointment request. The appointment request link is also fixed and issue found in scenario when appointment request for clinic is enabled, user click link to request appointment, and then user clicks back button. Before fix, this scenario would result in a button with a spinning icon and an error logged to the console.

**NOTE:** The issue of 2 alert messages being displayed will be address in another ticket

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#51172

## Testing done
Unit and e2e testing

## How to test
To test this fix you will need to simulate a network error when retrieving appointment slots. Then can be done in Chrome developer tools by:

1. Scheduling an appointment like normal up to the 'Choose date and time' page
2. Go to 'Network' tab
3. Find and select the 'http://localhost:3000/vaos/v2/locations/983/clinics/455/slots?start=2023-01-01T00:00:00-06:00&end=2023-02-28T00:00:00-06:00' network request
4. Right click on this request and select 'Block request url'. This simulates the error
5. Refresh page and schedule the appointment again
6. You should see the error message when you get to the 'Choose date and time' page
7. Verify that the correct alert message is displayed for each scenario:
    - Clinic is configured for appointment request
    - Clinic is configured for appointment request and 'submit a request for a VA appointment' link is clicked
    - Clinic is configured for appointment request, 'submit a request for a VA appointment' link is clicked and back button is clicked on 'Choose and appointment day and time' page
    - Clinic is not configured for appointment request

**NOTE:** Clinic configuration for V2 is controlled via ./vaos/services/mocks/V2/scheduling_configuration.json. Find the location id, find the type of care, toggle the request.enabled value

## Screenshots
**Clinic configured to accept appointment request**
---
localhost_3001_health-care_schedule-view-va-appointments_appointments_new-appointment_preferred-date

**Clinic not configured to accept appointment request**
---
localhost_3001_health-care_schedule-view-va-appointments_appointments_new-appointment_select-date

## What areas of the site does it impact?
VAOS

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/vaos/new-appointment/components/DateTimeSelectPage/index.jsx [hunk 1]
```diff
@@ -89,22 +123,26 @@ export default function DateTimeSelectPage() {
...
+          moment(preferredDate)
+            .add(1, 'months')
+            .endOf('month')
+            .format('YYYY-MM-DD'),
+          true,
+        ),
+      );
+      document.title = `${pageTitle} | Veterans Affairs`;
+    },
+    [dispatch, preferredDate],
```
Comment (vbahinwillit) [inline src/applications/vaos/new-appointment/components/DateTimeSelectPage/index.jsx:144]: Fix esLint waring

**Review by simiadebowale — APPROVED** [review]
Tested, and 'submit a request for a VA appointment' link is working as expected

File: src/applications/vaos/new-appointment/components/DateTimeSelectPage/index.jsx [hunk 2]
```diff
@@ -52,10 +48,48 @@ function ErrorMessage({ facilityId, history }) {
...
+      aria-atomic="true"
+      aria-live="assertive"
+      className="vads-u-margin-bottom--2"
+    >
+      <InfoAlert
+        status="error"
+        level="2"
+        headline="We’ve run into a problem trying to find an appointment time"
+      >
+        {renderContent({ dispatch, isRequest, facilityId, history })}
```
Comment (jenniemc) [inline src/applications/vaos/new-appointment/components/DateTimeSelectPage/index.jsx:83]: Typically when we want the specific property from an object we write it like 
`const { request } = eligibility`

What is the advantage of writing out `const { request: isRequest } = eligibility` ?

**Review by jenniemc — APPROVED** [review]
Looks good to me! Just have one question

File: src/applications/vaos/new-appointment/components/DateTimeSelectPage/index.jsx (hunk 2 above)
Comment (vbahinwillit) [reply src/applications/vaos/new-appointment/components/DateTimeSelectPage/index.jsx:83]: The property is being renamed too 'isRequest'. I prefer 'is' prefix for boolean variables.

**Review by rmessina1010 — APPROVED** [review]
LGTM

**Pull request merged**